dask/base.py

```diff
@@ -1605,4 +1608,51 @@ def clone_key(key, seed):
     raise TypeError(f"Expected str or tuple[str, Hashable, ...]; got {key}")
 
 
+def print_on_compute(msg: str, val: T) -> T:
+    """When the value is computed, print it using distributed.print.
+
+    Note: this may affect performance, since it will change the graph being
+    computed.
+
+    Parameters
+    ----------
+    msg : str
+          Format string for printing out value. Default "%s".
+          It's also possible to use placeholders compatible with str.format(),
+          such as "{}" or "{0}", or the named placeholder "{result}"
+    val : T
+          Lazy value, including a Dask collection, Scalar, etc.
+
+    Returns
+    -------
+    result : T
+          Original value wrapped with an additional Delayed layer
+    """
+    msg = msg or "%s"
+
+    import builtins
+
+    from dask import delayed
+
+    first_key = next(flatten(val.__dask_keys__()))  # type: ignore
+    val_name = get_name_from_key(first_key)
+    dask_key_name = f"{val_name}-print"
+
+    @delayed  # type: ignore
+    def print_future(result) -> T:
+        try:
+            from distributed.worker import print
+        except ImportError:
+            print = builtins.print
+        if "{result" in msg:
+            print(msg.format(result=result))
+        elif "%" in msg:
+            print(msg % result)
```

Comment:
This is broken:
`print_on_compute("The percentage is {}%")`

This whole logic feels complicated and unneeded.
I think that most users would be comfortable if you replicated the API of logging.info, which exclusively accepts % formatting: `logging.info("I have %d apples", n_apples)`

Reply:
Ok.